lib/ibmdb.js

```diff
@@ -409,32 +425,37 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
       // Execution for updateOrCreate requires running two
       // separate SQL statements.  The second depends on the
       // result of the first.
+      var where = {};
+      where[id] = data[id];
+
       var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
       countStmt.merge(tableName);
-      countStmt.merge(self.buildWhere(data));
+      countStmt.merge(self.buildWhere(model, where));
+      countStmt.noResults = false;
+
+      connection.query(countStmt, function(err, countData) {
+        if (err) {
+          debug('IBMDB.prototype.updateOrCreate (ERROR):%j', err);
+          return cb(err);
+        }
```

Comment:
```
if(err) return cb(err);
```
Is what we usually use in LoopBack; I don't think if you need to use debug for error